irods/manager/data_object_manager.py

```diff
@@ -84,29 +84,44 @@ def should_parallelize_transfer( self,
                                      num_threads = 0,
                                      obj_sz = 1+MAXIMUM_SINGLE_THREADED_TRANSFER_SIZE,
                                      server_version_hint = (),
-                                     measured_obj_size = ()  ## output variable. If a list is provided, it shall
-                                                              # be truncated to contain one value, the size of the
-                                                              # seekable object (if one is provided for `obj_sz').
+                                     measured_obj_size = (), ## output variable. If a list is provided, it shall
+                                                              # be truncated to contain one value, the presumed size of the
+                                                              # file or data object - if one is provided for `obj_sz'.  That size is
+                                                              # determined normally by stat'ing (using a seek-to-end operation) but
+                                                              # can be overridden using the DATA_SIZE_KW option.
+                                     open_options = ()        # If this parameter is of type dict, set the DATA_SIZE_KW item from
+                                                              # the stat'ed size
         ):
 
+        provided_data_size = int(dict(open_options).get(kw.DATA_SIZE_KW,"-1"))
```

Comment:
need to sniff the server version here too?

not sure, just seeing the hard-coded `-1`.

Reply:
We may need to review how DATA_SIZE_KW is important.  As written, it's used (probably by S3) to dictate what filesize to use if the file to be put is in S3 and therefore we'd like not to have to stat it.  
Also, if DATA_SIZE_KW is set explicitly to "-1" in servers where the default dataSize needs to  be sent as a zero, how should we handle that?
I admit there are too many questions in my mind to intelligently implement this at present.

Reply:
Also.  I recall there was some proviso about the dataSize and its semantic/significance when OPENing multiple replicas for parallel put.   That exact semantic has slipped my mind, so I may need to be better apprised of that, as well.  Good topic for discuss.

Reply:
This has been updated. Resolving.